packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_type3_staticprofile.cpp

```diff
@@ -0,0 +1,322 @@
+  //   
+  auto domain_map = row_map;
+  auto range_map  = row_map;
+
+  auto owned_element_to_node_ids = mesh.getOwnedElementToNode();
+  auto ghost_element_to_node_ids = mesh.getGhostElementToNode();
+
+  RCP<TimeMonitor> timerGlobal = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("X) Global")));
+  RCP<TimeMonitor> timerElementLoopGraph = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("1) ElementLoop  (Graph)")));
+
+  RCP<GraphType> crs_graph = rcp(new GraphType(row_map, maxEntriesPerRow, Tpetra::StaticProfile));
+  
+  // Using 4 because we're using quads for this example, so there will be 4 nodes associated with each element.
+  Teuchos::Array<GlobalOrdinal> global_ids_in_row(4);
+
+  // Insert node contributions for every OWNED element:
+  for(size_t element_gidx=0; element_gidx<mesh.getNumOwnedElements(); element_gidx++)
+  {
+    // Populate global_ids_in_row:
+    // - Copy the global node ids for current owned element into an array.  
+    // - Since each element's contribution is a clique, we can re-use this for 
+    //   each row associated with this element's contribution.
+    for(size_t element_node_idx=0; element_node_idx<owned_element_to_node_ids.extent(1); element_node_idx++)
+    {
+      global_ids_in_row[element_node_idx] = owned_element_to_node_ids(element_gidx, element_node_idx);
+    }
+
+    // Add the contributions from the current row into the graph if the node is owned.
+    // - For example, if Element 0 contains nodes [0,1,4,5] and nodes 0 and 4 are owned
+    //   by the current processor, then:
+    //   - node 0 inserts [0, 1, 4, 5]
+    //   - node 1 <skip>
+    //   - node 4 inserts [0, 1, 4, 5]
+    //   - node 5 <skip>
+    for(size_t element_node_idx=0; element_node_idx<owned_element_to_node_ids.extent(1); element_node_idx++)
+    {
+      if(mesh.nodeIsOwned(global_ids_in_row[element_node_idx]))
+      {
+       crs_graph->insertGlobalIndices(global_ids_in_row[element_node_idx], global_ids_in_row());
+      }
+    }
+  }
+
+  // Insert the node contributions for every GHOST element:
+  for(size_t element_gidx=0; element_gidx<mesh.getNumGhostElements(); element_gidx++)
+  {
+    for(size_t element_node_idx=0; element_node_idx<ghost_element_to_node_ids.extent(1); element_node_idx++)
+    {
+      global_ids_in_row[element_node_idx] = ghost_element_to_node_ids(element_gidx, element_node_idx);
+    }
+    for(size_t element_node_idx=0; element_node_idx<ghost_element_to_node_ids.extent(1); element_node_idx++)
+    {
+      if(mesh.nodeIsOwned(global_ids_in_row[element_node_idx]))
+      {
+       crs_graph->insertGlobalIndices(global_ids_in_row[element_node_idx], global_ids_in_row());
+      }
+    }
+  }
+
+  timerElementLoopGraph = Teuchos::null;
+
+  // 'finalize' the crs_graph by calling fillComplete().
+  {
+    RCP<TimeMonitor> timerFillCompleteGraph = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("2) FillComplete (Graph)")));
+    crs_graph->fillComplete();
```

Comment:
It's nicer to pass in the domain and range Map here; it reminds users to do this, even when they might think it's not necessary.

Reply:
@mhoemmen Hmmm... the TotalElementLoop_SP example (formerly Type-3 SP) does not have both a domain and range map in the application.  I have the `row_map` though... is that one the range or domain?  what about the other?  use it for both parameters?